### ISSUE TITLE
Importerror from Zendesk

### ISSUE THREAD

**Issue by danielduchon** [zammad/zammad#2179]
### Infos:

* Used Zammad version: 2.5
* Installation method (source, package, ..): DEB
* Operating system: Ubuntu 16.04
* Database + version: 9.5
* Elasticsearch version: 5.6.10
* Browser + version: chrome 67.0.3396.99

### Expected behavior:

* import all data without error

### Actual behavior:

* zammad imports only the first 250 characters with error:
`WARNING: cut string because of database length Ticket.note(250 but is 2780:`
* after 523 tickets the import stops with the following error message:
```
E, [2018-08-09T13:09:52.192645 #11035] ERROR -- : #<Net::OpenTimeout: execution expired>
I, [2018-08-09T13:09:52.193324 #11035]  INFO -- : Sequence 'Sequencer::Sequence::Import::Zendesk::Ticket::Comment::Attachment' Unit 'Sequencer::Unit::Import::Zendesk::Ticket::Comment::Attachment::Request' (index: 0) finished.
I, [2018-08-09T13:09:52.193738 #11035]  INFO -- : State cleanup of Unit Sequencer::Unit::Import::Zendesk::Ticket::Comment::Attachment::Request (index: 0) started.
I, [2018-08-09T13:09:52.194007 #11035]  INFO -- : State cleanup of Unit Sequencer::Unit::Import::Zendesk::Ticket::Comment::Attachment::Request (index: 0) finished.
I, [2018-08-09T13:09:52.194321 #11035]  INFO -- : Sequence 'Sequencer::Sequence::Import::Zendesk::Ticket::Comment::Attachment' Unit 'Sequencer::Unit::Import::Zendesk::Ticket::Comment::Attachment::Add' (index: 1) started.
E, [2018-08-09T13:09:52.195032 #11035] ERROR -- : undefined method 'body' for nil:NilClass (NoMethodError)
/opt/zammad/lib/sequencer/unit/import/zendesk/ticket/comment/attachment/add.rb:20:in 'process'
/opt/zammad/lib/sequencer/unit/import/common/model/mixin/skip/action.rb:45:in 'process'
/opt/zammad/lib/sequencer/unit/base.rb:200:in 'process'
/opt/zammad/lib/sequencer.rb:65:in 'block (4 levels) in process'
/opt/zammad/lib/mixin/start_finish_logger.rb:7:in 'log_start_finish'
/opt/zammad/lib/sequencer.rb:64:in 'block (3 levels) in process'
/opt/zammad/lib/sequencer/state.rb:149:in 'process'
/opt/zammad/lib/sequencer.rb:62:in 'block (2 levels) in process'
/opt/zammad/lib/sequencer/units.rb:29:in 'block in each'
/opt/zammad/lib/sequencer/units.rb:28:in 'each'
/opt/zammad/lib/sequencer/units.rb:28:in 'each'
/opt/zammad/lib/sequencer.rb:60:in 'each_with_index'
/opt/zammad/lib/sequencer.rb:60:in 'block in process'
/opt/zammad/lib/mixin/start_finish_logger.rb:7:in 'log_start_finish'
/opt/zammad/lib/sequencer.rb:58:in 'process'
/opt/zammad/lib/sequencer.rb:25:in 'process'
/opt/zammad/lib/sequencer/unit/import/zendesk/sub_sequence/sub_object.rb:14:in 'block in process'
/opt/zammad/lib/sequencer/unit/import/zendesk/sub_sequence/base.rb:41:in 'each'
/opt/zammad/lib/sequencer/unit/import/zendesk/sub_sequence/base.rb:41:in 'public_send'
/opt/zammad/lib/sequencer/unit/import/zendesk/sub_sequence/base.rb:41:in 'resource_iteration'
/opt/zammad/lib/sequencer/unit/import/zendesk/sub_sequence/sub_object.rb:12:in 'process'
/opt/zammad/lib/sequencer/unit/import/zendesk/ticket/comment/attachments.rb:13:in 'process'
/opt/zammad/lib/sequencer/unit/base.rb:200:in 'process'
/opt/zammad/lib/sequencer.rb:65:in 'block (4 levels) in process'
/opt/zammad/lib/mixin/start_finish_logger.rb:7:in 'log_start_finish'
/opt/zammad/lib/sequencer.rb:64:in 'block (3 levels) in process'
/opt/zammad/lib/sequencer/state.rb:149:in 'process'
/opt/zammad/lib/sequencer.rb:62:in 'block (2 levels) in process'
/opt/zammad/lib/sequencer/units.rb:29:in 'block in each'
/opt/zammad/lib/sequencer/units.rb:28:in 'each'
/opt/zammad/lib/sequencer/units.rb:28:in 'each'
/opt/zammad/lib/sequencer.rb:60:in 'each_with_index'
/opt/zammad/lib/sequencer.rb:60:in 'block in process'
/opt/zammad/lib/mixin/start_finish_logger.rb:7:in 'log_start_finish'
/opt/zammad/lib/sequencer.rb:58:in 'process'
/opt/zammad/lib/sequencer.rb:25:in 'process'
/opt/zammad/lib/sequencer/unit/import/zendesk/sub_sequence/sub_object.rb:14:in 'block in process'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/zendesk_api-1.16.0/lib/zendesk_api/collection.rb:349:in 'block in _all'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/zendesk_api-1.16.0/lib/zendesk_api/collection.rb:457:in 'each'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/zendesk_api-1.16.0/lib/zendesk_api/collection.rb:457:in 'public_send'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/zendesk_api-1.16.0/lib/zendesk_api/collection.rb:457:in 'array_method'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/zendesk_api-1.16.0/lib/zendesk_api/collection.rb:298:in 'method_missing'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/zendesk_api-1.16.0/lib/zendesk_api/collection.rb:342:in '_all'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/zendesk_api-1.16.0/lib/zendesk_api/collection.rb:216:in 'all!'
/opt/zammad/lib/sequencer/unit/import/zendesk/sub_sequence/base.rb:41:in 'public_send'
/opt/zammad/lib/sequencer/unit/import/zendesk/sub_sequence/base.rb:41:in 'resource_iteration'
/opt/zammad/lib/sequencer/unit/import/zendesk/tickets.rb:21:in 'resource_iteration'
/opt/zammad/lib/sequencer/unit/import/zendesk/sub_sequence/mapped.rb:45:in 'process_sub_sequence'
/opt/zammad/lib/sequencer/unit/import/zendesk/sub_sequence/mapped.rb:29:in 'block in process'
/opt/zammad/lib/sequencer/state.rb:45:in 'provide'
/opt/zammad/lib/sequencer/unit/import/zendesk/sub_sequence/mapped.rb:28:in 'process'
/opt/zammad/lib/sequencer/unit/base.rb:200:in 'process'
/opt/zammad/lib/sequencer.rb:65:in 'block (4 levels) in process'
/opt/zammad/lib/mixin/start_finish_logger.rb:7:in 'log_start_finish'
/opt/zammad/lib/sequencer.rb:64:in 'block (3 levels) in process'
/opt/zammad/lib/sequencer/state.rb:149:in 'process'
/opt/zammad/lib/sequencer.rb:62:in 'block (2 levels) in process'
/opt/zammad/lib/sequencer/units.rb:29:in 'block in each'
/opt/zammad/lib/sequencer/units.rb:28:in 'each'
/opt/zammad/lib/sequencer/units.rb:28:in 'each'
/opt/zammad/lib/sequencer.rb:60:in 'each_with_index'
/opt/zammad/lib/sequencer.rb:60:in 'block in process'
/opt/zammad/lib/mixin/start_finish_logger.rb:7:in 'log_start_finish'
/opt/zammad/lib/sequencer.rb:58:in 'process'
/opt/zammad/lib/sequencer.rb:25:in 'process'
/opt/zammad/lib/import/mixin/sequence.rb:15:in 'process'
/opt/zammad/lib/import/zendesk.rb:8:in 'start'
/opt/zammad/app/models/import_job.rb:24:in 'start'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/delayed_job-4.1.3/lib/delayed/performable_method.rb:26:in 'perform'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/delayed_job-4.1.3/lib/delayed/backend/base.rb:81:in 'block in invoke_job'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/delayed_job-4.1.3/lib/delayed/lifecycle.rb:61:in 'block in initialize'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/delayed_job-4.1.3/lib/delayed/lifecycle.rb:66:in 'execute'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/delayed_job-4.1.3/lib/delayed/lifecycle.rb:40:in 'run_callbacks'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/delayed_job-4.1.3/lib/delayed/backend/base.rb:78:in 'invoke_job'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/delayed_job-4.1.3/lib/delayed/worker.rb:230:in 'block (2 levels) in run'
/opt/zammad/vendor/ruby-2.4.4/lib/ruby/2.4.0/timeout.rb:93:in 'block in timeout'
/opt/zammad/vendor/ruby-2.4.4/lib/ruby/2.4.0/timeout.rb:103:in 'timeout'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/delayed_job-4.1.3/lib/delayed/worker.rb:230:in 'block in run'
/opt/zammad/vendor/ruby-2.4.4/lib/ruby/2.4.0/benchmark.rb:308:in 'realtime'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/delayed_job-4.1.3/lib/delayed/worker.rb:229:in 'run'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/delayed_job-4.1.3/lib/delayed/worker.rb:312:in 'block in reserve_and_run_one_job'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/delayed_job-4.1.3/lib/delayed/lifecycle.rb:61:in 'block in initialize'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/delayed_job-4.1.3/lib/delayed/lifecycle.rb:66:in 'execute'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/delayed_job-4.1.3/lib/delayed/lifecycle.rb:40:in 'run_callbacks'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/delayed_job-4.1.3/lib/delayed/worker.rb:312:in 'reserve_and_run_one_job'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/delayed_job-4.1.3/lib/delayed/worker.rb:213:in 'block in work_off'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/delayed_job-4.1.3/lib/delayed/worker.rb:212:in 'times'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/delayed_job-4.1.3/lib/delayed/worker.rb:212:in 'work_off'
/opt/zammad/app/models/scheduler.rb:355:in 'block (3 levels) in worker'
/opt/zammad/vendor/ruby-2.4.4/lib/ruby/2.4.0/benchmark.rb:308:in 'realtime'
/opt/zammad/app/models/scheduler.rb:353:in 'block (2 levels) in worker'
/opt/zammad/app/models/scheduler.rb:349:in 'loop'
/opt/zammad/app/models/scheduler.rb:349:in 'block in worker'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/logging-2.2.2/lib/logging/diagnostic_context.rb:474:in 'block in create_with_logging_context'
```
### Steps to reproduce the behavior:

* setup a Zendesk-instanz and try to import data from it to zammad.

Yes I'm sure this is a bug and no feature request or a general question.

**Comment by thorsteneckel** [comment]
Hi @danielduchon - there was an connection timeout while downloading an attachment from Zendesk. This might be the case for lager attachments or if requests to Zendesk take to long. Do you have tried to restart the import from scratch?
However, Zammad should handle this more properly. We'll handle this.

From my first analysis:
Zammad [checks if the download request was successful](https://github.com/zammad/zammad/blob/develop/lib/sequencer/unit/import/zendesk/ticket/comment/attachment/request.rb#L27). If not it logs the error (which might be the cause for the `#<Net::OpenTimeout: execution expired>` line in the log). For failed requests the `skipped` action attribute [is set](https://github.com/zammad/zammad/blob/develop/lib/sequencer/unit/import/zendesk/ticket/comment/attachment/request.rb#L18). Which [causes the next step/unit to be skipped](https://github.com/zammad/zammad/blob/develop/lib/sequencer/unit/import/zendesk/ticket/comment/attachment/add.rb#L12).
This doesn't work as expected. The unit is not skipped.

**Comment by danielduchon** [comment]
hi @thorsteneckel sorry for my delayed answer. yes, I tried it several times and it stops every time at the same ticket. I can't see anything special in it. there are only two little images with size off 6kb together.

**Comment by danielduchon** [comment]
wrong button...

**Comment by SummerStorm** [comment]
Hi there, we recently increased the Zendesk attachment import timeout on `develop`. Could you please try the latest `develop` if you are still experiencing the same issue?

**Comment by MrGeneration** [comment]
@danielduchon do you still have your old Zendesk instance for testing?

**Comment by thorsteneckel** [comment]
From a technical perspective I can say that this is the issue that was fixed with #2440 - thanks again to @anuzhin 😍 Closing now. @danielduchon  Feel free to leave feedback if doesn't solve your issue. Thanks!
